test/files/pos/classOfObjectType/Foo.scala

```diff
@@ -0,0 +1,7 @@
+
+object Bar
+
+trait Foo {
+  @AnnotationWithClassType(cls = classOf[Bar.type])
+  def function: Any = ???
+}
```

Comment:
I added this compile test because the issue shows up when using a Java annotation that requires a class. But of course, the impl allows many other usages.

When I faced that issue for the first time, I tried to used it as follows:

```scala
trait Foo {
  @AnnotationWithClassType(cls = Bar.getClass)
  def function: Any = ???
}
```

which gives the following error:

```scala
Foo.scala:5: error: annotation argument needs to be a constant; found: Bar.getClass()
  @AnnotationWithClassType(cls = Bar.getClass)
```